### ISSUE TITLE
[Pico][!NSFW]Panoramic Video play with sound only

### ISSUE THREAD

**Issue by picoxr** [MozillaReality/FirefoxReality#3191]
## Configuration

Firefox Reality version: Firefox Reality version 10-141060720
Firefox Reality build ID: 9b36c14

Hardware: 
Pico G2 4K - Android - 8.1.0

## Steps to Reproduce

1. Open web video  Caution NSFW content "https://npx-webvtt2.duckdns.org/vr/180.html"
2. Click play button
3.

## Current Behavior

Only audio is heard but no video images displayed.

## Expected Behavior

Both Video and Audio can be output correctly.

## Context

Panoramic video web player

## Error Logs and Stack Traces

<details open>

```
```

</details>

**Comment by bluemarvin** [comment]
I'm seeing this in the logcat:
```
04-29 13:47:11.548  4997  5153 W Web Content: [JavaScript Warning: "WebGL warning: drawArraysInstanced: TEXTURE_2D at unit 0 is incomplete: The dimensions of `level_base` are not all positive." {file: "https://cdn.jsdelivr.net/npm/videojs-vr@1.7.2/dist/videojs-vr.js" line: 18184}]
04-29 13:47:11.550  4997  5153 W Web Content: [JavaScript Warning: "WebGL warning: drawArraysInstanced: TEXTURE_2D at unit 0 is incomplete: The dimensions of `level_base` are not all positive." {file: "https://cdn.jsdelivr.net/npm/videojs-vr@1.7.2/dist/videojs-vr.js" line: 18184}]
04-29 13:47:11.552  4997  5153 W Web Content: [JavaScript Warning: "WebGL warning: drawArraysInstanced: TEXTURE_2D at unit 0 is incomplete: The dimensions of `level_base` are not all positive." {file: "https://cdn.jsdelivr.net/npm/videojs-vr@1.7.2/dist/videojs-vr.js" line: 18184}]
04-29 13:47:11.554  4997  5153 W Web Content: [JavaScript Warning: "WebGL warning: drawArraysInstanced: TEXTURE_2D at unit 0 is incomplete: The dimensions of `level_base` are not all positive." {file: "https://cdn.jsdelivr.net/npm/videojs-vr@1.7.2/dist/videojs-vr.js" line: 18184}]
04-29 13:47:11.560  4997  5153 W Web Content: [JavaScript Warning: "WebGL warning: drawArraysInstanced: TEXTURE_2D at unit 0 is incomplete: The dimensions of `level_base` are not all positive." {file: "https://cdn.jsdelivr.net/npm/videojs-vr@1.7.2/dist/videojs-vr.js" line: 18184}]
04-29 13:47:11.576  4997  5153 W Web Content: [JavaScript Warning: "WebGL warning: drawArraysInstanced: TEXTURE_2D at unit 0 is incomplete: The dimensions of `level_base` are not all positive." {file: "https://cdn.jsdelivr.net/npm/videojs-vr@1.7.2/dist/videojs-vr.js" line: 18184}]
04-29 13:47:11.579  4997  5153 W Web Content: [JavaScript Warning: "WebGL warning: drawArraysInstanced: TEXTURE_2D at unit 0 is incomplete: The dimensions of `level_base` are not all positive." {file: "https://cdn.jsdelivr.net/npm/videojs-vr@1.7.2/dist/videojs-vr.js" line: 18184}]
04-29 13:47:11.581  4997  5153 W Web Content: [JavaScript Warning: "WebGL warning: drawArraysInstanced: TEXTURE_2D at unit 0 is incomplete: The dimensions of `level_base` are not all positive." {file: "https://cdn.jsdelivr.net/npm/videojs-vr@1.7.2/dist/videojs-vr.js" line: 18184}]
04-29 13:47:11.584  4997  5153 W Web Content: [JavaScript Warning: "WebGL warning: drawArraysInstanced: TEXTURE_2D at unit 0 is incomplete: The dimensions of `level_base` are not all positive." {file: "https://cdn.jsdelivr.net/npm/videojs-vr@1.7.2/dist/videojs-vr.js" line: 18184}]
04-29 13:47:11.589  4997  5153 W Web Content: [JavaScript Warning: "WebGL warning: drawArraysInstanced: TEXTURE_2D at unit 0 is incomplete: The dimensions of `level_base` are not all positive." {file: "https://cdn.jsdelivr.net/npm/videojs-vr@1.7.2/dist/videojs-vr.js" line: 18184}]
04-29 13:47:11.591  4997  5153 W Web Content: [JavaScript Warning: "WebGL warning: drawArraysInstanced: TEXTURE_2D at unit 0 is incomplete: The dimensions of `level_base` are not all positive." {file: "https://cdn.jsdelivr.net/npm/videojs-vr@1.7.2/dist/videojs-vr.js" line: 18184}]
04-29 13:47:11.596  4997  5153 W Web Content: [JavaScript Warning: "WebGL warning: drawArraysInstanced: TEXTURE_2D at unit 0 is incomplete: The dimensions of `level_base` are not all positive." {file: "https://cdn.jsdelivr.net/npm/videojs-vr@1.7.2/dist/videojs-vr.js" line: 18184}]
04-29 13:47:11.599  4997  5153 W Web Content: [JavaScript Warning: "WebGL warning: drawArraysInstanced: TEXTURE_2D at unit 0 is incomplete: The dimensions of `level_base` are not all positive." {file: "https://cdn.jsdelivr.net/npm/videojs-vr@1.7.2/dist/videojs-vr.js" line: 18184}]
04-29 13:47:11.606  4997  5153 W Web Content: [JavaScript Warning: "WebGL warning: drawArraysInstanced: TEXTURE_2D at unit 0 is incomplete: The dimensions of `level_base` are not all positive." {file: "https://cdn.jsdelivr.net/npm/videojs-vr@1.7.2/dist/videojs-vr.js" line: 18184}]
04-29 13:47:11.618  4997  5153 W Web Content: [JavaScript Warning: "WebGL warning: drawArraysInstanced: TEXTURE_2D at unit 0 is incomplete: The dimensions of `level_base` are not all positive." {file: "https://cdn.jsdelivr.net/npm/videojs-vr@1.7.2/dist/videojs-vr.js" line: 18184}]
04-29 13:47:11.621  4997  5153 W Web Content: [JavaScript Warning: "WebGL warning: drawArraysInstanced: TEXTURE_2D at unit 0 is incomplete: The dimensions of `level_base` are not all positive." {file: "https://cdn.jsdelivr.net/npm/videojs-vr@1.7.2/dist/videojs-vr.js" line: 18184}]
04-29 13:47:11.631  4997  5153 W Web Content: [JavaScript Warning: "WebGL warning: drawArraysInstanced: TEXTURE_2D at unit 0 is incomplete: The dimensions of `level_base` are not all positive." {file: "https://cdn.jsdelivr.net/npm/videojs-vr@1.7.2/dist/videojs-vr.js" line: 18184}]
04-29 13:47:11.644  4997  5153 W Web Content: [JavaScript Warning: "WebGL warning: drawArraysInstanced: TEXTURE_2D at unit 0 is incomplete: The dimensions of `level_base` are not all positive." {file: "https://cdn.jsdelivr.net/npm/videojs-vr@1.7.2/dist/videojs-vr.js" line: 18184}]
04-29 13:47:11.658  4997  5153 W Web Content: [JavaScript Warning: "WebGL warning: drawArraysInstanced: TEXTURE_2D at unit 0 is incomplete: The dimensions of `level_base` are not all positive." {file: "https://cdn.jsdelivr.net/npm/videojs-vr@1.7.2/dist/videojs-vr.js" line: 18184}]
04-29 13:47:11.671  4997  5153 W Web Content: [JavaScript Warning: "WebGL warning: drawArraysInstanced: TEXTURE_2D at unit 0 is incomplete: The dimensions of `level_base` are not all positive." {file: "https://cdn.jsdelivr.net/npm/videojs-vr@1.7.2/dist/videojs-vr.js" line: 18184}]
04-29 13:47:11.688  4997  5153 W Web Content: [JavaScript Warning: "WebGL warning: drawArraysInstanced: TEXTURE_2D at unit 0 is incomplete: The dimensions of `level_base` are not all positive." {file: "https://cdn.jsdelivr.net/npm/videojs-vr@1.7.2/dist/videojs-vr.js" line: 18184}]
04-29 13:47:11.699  4997  5153 W Web Content: [JavaScript Warning: "WebGL warning: drawArraysInstanced: TEXTURE_2D at unit 0 is incomplete: The dimensions of `level_base` are not all positive." {file: "https://cdn.jsdelivr.net/npm/videojs-vr@1.7.2/dist/videojs-vr.js" line: 18184}]
04-29 13:47:11.714  4997  5153 W Web Content: [JavaScript Warning: "WebGL warning: drawArraysInstanced: TEXTURE_2D at unit 0 is incomplete: The dimensions of `level_base` are not all positive." {file: "https://cdn.jsdelivr.net/npm/videojs-vr@1.7.2/dist/videojs-vr.js" line: 18184}]
04-29 13:47:11.727  4997  5153 W Web Content: [JavaScript Warning: "WebGL warning: drawArraysInstanced: TEXTURE_2D at unit 0 is incomplete: The dimensions of `level_base` are not all positive." {file: "https://cdn.jsdelivr.net/npm/videojs-vr@1.7.2/dist/videojs-vr.js" line: 18184}]
04-29 13:47:11.742  4997  5153 W Web Content: [JavaScript Warning: "WebGL warning: drawArraysInstanced: TEXTURE_2D at unit 0 is incomplete: The dimensions of `level_base` are not all positive." {file: "https://cdn.jsdelivr.net/npm/videojs-vr@1.7.2/dist/videojs-vr.js" line: 18184}]
04-29 13:47:11.756  4997  5153 W Web Content: [JavaScript Warning: "WebGL warning: drawArraysInstanced: TEXTURE_2D at unit 0 is incomplete: The dimensions of `level_base` are not all positive." {file: "https://cdn.jsdelivr.net/npm/videojs-vr@1.7.2/dist/videojs-vr.js" line: 18184}]
04-29 13:47:11.756  4997  5153 W Web Content: [JavaScript Warning: "WebGL: No further warnings will be reported for this WebGL context. (already reported 32 warnings)" {file: "https://cdn.jsdelivr.net/npm/videojs-vr@1.7.2/dist/videojs-vr.js" line: 18184}]
```

**Comment by bluemarvin** [comment]
@fernandojsg This seems to be three.js based. Is there a good way to debug what is not working in FxR?

**Comment by fernandojsg** [comment]
@bluemarvin the error seems to be related to the video used as a texture itself. It's using a .m3u8 "playlist" and it doesn't seem to be handled correctly on geckoview. I tried on Firefox for Android and it doesn't work either (Just blank video even without entering VR).

On Firefox desktop you can see how it first request the .m3u8 playlist
![image](https://user-images.githubusercontent.com/782511/80704741-67182180-8ae5-11ea-8648-d1bd4c4d0eb3.png)
And then it will fetch a second .m3u8 with the fragments and it just keep requesting the `.ts` with the actual video frames used as a texture.

but on android/FxR it gets stuck on the first `m3u8` request and it's shown initially as 206 partial content.

Looking around google I found some threads of people complaining about problems when dealing with streaming `m3u8` on firefox, some older some newer, so I guess it's something related to it and not related to three itself.

**Comment by fernandojsg** [comment]
It seems that the issue is related to streaming HLS video. I added some comments on https://bugzilla.mozilla.org/show_bug.cgi?id=1619550

**Comment by miketaylr** [comment]
Here's another issue related to three.js not working anymore: https://github.com/webcompat/web-bugs/issues/47721

**Comment by bluemarvin** [comment]
The bugzilla issue was resolved. We should retest.

**Comment by bluemarvin** [comment]
I've tested and it is fixed in latest nightly. We can close once we bump GV versions.

**Comment by bluemarvin** [comment]
fixed by #3553
